### PR TITLE
Add s3 and postera upload of artifacts. 

### LINKED ISSUE #352: Add push and pull for S3 and link to postera at end of end2end docking workflow.
Add push and pull to S3 for poses and GIFs and link to postera at end of end2end docking workflow.

### PR DESCRIPTION
**Pull request by hmacdope**
fixes #352 

Adds  s3 and postera upload of artifacts that can be viewed in manifold via the new cloudfront API.

### CODE REVIEW

**Comment by hmacdope** [comment]
@dotsdl would I be able to grab a review at some point?

**Comment by dotsdl** [comment]
Yes, reviewing today!

File: asapdiscovery-data/asapdiscovery/data/postera/manifold_artefacts.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    """
+
+    def __init__(
+        self,
+        molecule_dataframe: pd.DataFrame,
+        molecule_set_id: UUID,
+        artifact_type: ArtifactType,
+        moleculeset_api: MoleculeSetAPI,
+        cloud_front: CloudFront,
```
Comment (dotsdl) [inline asapdiscovery-data/asapdiscovery/data/postera/manifold_artefacts.py:38]: cloud_front -> cloudfront

File: asapdiscovery-dataviz/asapdiscovery/dataviz/gif_viz.py [hunk 2]
```diff
@@ -270,7 +270,8 @@ def write_traj_visualization(self, traj, system, path):
             show_contacts(p, "ligand", "receptor")
 
         p.cmd.set_view(self.view_coords)  # sets general orientation
-        # p.cmd.zoom("resn UNK", buffer=4)  # zoom to ligand
+        # NOTE: @JScheen, I added this back in because the view was somehow broken.
+        p.cmd.zoom("resn UNK", buffer=4)  # zoom to ligand
```
Comment (dotsdl) [inline asapdiscovery-dataviz/asapdiscovery/dataviz/gif_viz.py:274]: Note from @hmacdope : change this back

File: asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py [hunk 3]
```diff
@@ -381,10 +392,38 @@ def main():
...
-        logger.info(f"attempting to pull Molecule Set: {args.mols} from PostEra")
+        aws_access_key_id = os.getenv("AWS_ACCESS_KEY_ID")
+        if aws_access_key_id is None:
+            raise ValueError("Environment variable AWS_ACCESS_KEY_ID not found")
+
+        aws_secret_access_key = os.getenv("AWS_SECRET_ACCESS_KEY")
+        if aws_secret_access_key is None:
+            raise ValueError("Environment variable AWS_SECRET_ACCESS_KEY not found")
+
+        artefact_bucket_name = os.getenv("ARTEFACT_BUCKET_NAME")
```
Comment (dotsdl) [inline asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py:403]: ARTEFACT -> ARTIFACT

File: asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py [hunk 4]
```diff
@@ -1056,12 +1095,17 @@ def dask_gif_adaptor(traj, system, outpath):
...
+
+    # keep in the artefact column for the poses and MD gifs
+    renamed_top_posit = rename_output_columns_for_manifold(
         top_posit,
         args.target,
+        column_enums,
         manifold_validate=True,
-        allow=[DockingResultCols.LIGAND_ID.value],
-        szybki=args.szybki,
+        allow=[DockingResultCols.LIGAND_ID.value, "_outpath_pose", "_outpath_gif"],
```
Comment (dotsdl) [inline asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py:1108]: Add a note for why we keep "_outpath_pose" and "_outpath_gif".

**Review by dotsdl — APPROVED** [review]
Looks great @hmacdope! Approved, but please address notes before merging!

**Pull request merged**